### ISSUE TITLE
Ribbon can be triggered by overlay function and then difficult to bring back

### ISSUE THREAD

**Issue by raindropsfromsky** [APEbbers/FreeCAD-Ribbon#133]
The sidepanels in FC (Treeview, tasks view and Property view) remain in Autohide mode, and appear only when we move the mouse over them.

In a similar fashion, the entire Ribbon Toolbar should be in Autohide mode. When the user moves the mouse over that area, it should appear. Once the user clicks on a button, and the mouse pointer leaves the area, the ribbon should disappear.

In fact, this function should be more sophisticated, like how Windows 11 autohides the Task Manager toolbar: It should show the toolbar only if the mouse moves into the area at a certain speed. If the mouse is moved in slowly, then the Ribbon should not appear. This lets the user work in the area reserved for the ribbon.

This will free up a massive part of the screen for the actual modeling.

**Comment by APEbbers** [comment]
Hi,

In FC 1.0 you can toggle the overlay function by pressing F3. The ribbon
should dissapear to the topleft corner of the view window.
I have disabled the AutoShow function in FC1.0 because entering the Ribbon
with the mouse interferes with this overlay function an makes the ribbon
disappear.

You can click on the pinbutton to enable the AutoHide function. When you
click on a tab, the ribbon appears. When you leave the ribbon with the
mouse the ribbon fold again.

Personally, I dont want the ribbon to disapear completly like the panels
because it is a replacement for the toolbars and not an another panel. But
I've not figured out how the exclude the ribbon from the overlay function.

kind regards

**Comment by raindropsfromsky** [comment]
> In FC 1.0 you can toggle the overlay function by pressing F3. The ribbon should dissapear to the topleft corner of the view window.

I tried F3 (after hovering the mouse on the Ribbon UI), but the Ribbon UI disappeared without leaving a target area where I can click to make it visible again. Even the side panels have disappeared.
The screen is completely blank. Only the Status line is visible.
There are no menus, which I can use to make the UI elements visible. 

I tried to press F3 to toggle the panels on, but the panels do not appear at all.

Restarting FC made no difference.

How do I restore the full UI?

**Comment by APEbbers** [comment]
in the topleft area of the viewport should be an target area. If you have a
panel left, it will be on the top right of that panel

**Comment by APEbbers** [comment]
Sorry for my late response, I just came back from a work dinner.

Did you succeed? Below a screenshot from my laptop where the area should be:
![image](https://github.com/user-attachments/assets/3fc9dc13-8fa8-496d-9c77-e67b0230a801)

**Comment by raindropsfromsky** [comment]
I don't see any "overflow" menu (three dots).

I see a completely clean screen.
![image](https://github.com/user-attachments/assets/d0aae6d6-69ac-41c3-a574-99baa341fc9f)

**Comment by raindropsfromsky** [comment]
If I have to reinstall FC, I'd like to save the Ribbon WB's settings which I had created so painstakingly.
How do I backup and restore the settings?

**Comment by APEbbers** [comment]
Before you reinstall FC can you try one of these buttons?
![image](https://github.com/user-attachments/assets/4ff52933-e554-43fd-83f5-ecb241b05f42)

The backup files and json files are located in the folder of the addon. under windows this is usually: 
![image](https://github.com/user-attachments/assets/915234ea-8c68-4b69-b6ef-3102a5b4067a)

**Comment by raindropsfromsky** [comment]
Thanks for the tip. 

I tried all the hotkeys one by one, but nothing works. 
The display remains completely blank.

---------------
I found the folder:
![image](https://github.com/user-attachments/assets/fa097717-155d-42c4-b2b0-42c044c8aca1)

Which is the backup file?
Also, what are the steps to save this file and restore it in the fresh vanilla installation to get the settings I crearted?

**Comment by APEbbers** [comment]
RibbonStructure is the settings file. The easiest way is to copy this file to your desktop.
After reinstall, you can place it back.

It might be better to copy the entire "mod" folder with you installed workbenches. If you place it back after reinstall,
All the workbench will be back. Saves you probably a lot of time installing add-ons

Normally there should be a folder named "backup". But I think the option for backups is not set yet.

**Comment by raindropsfromsky** [comment]
> Normally there should be a folder named "backup". But I think the option for backups is not set yet.

In Ribbon WB, I had set the backup path to a local folder.
It has created a 3.66 MB folder there.

How do I use this folder to restore the older settings?

(Or should I simply copy the entire "Mod" folder as you replied earlier?)
With that approach, I don't have to reinstall all WBs, right?

But then does the "Mod" folder contaion the Ribbon wB settings also?
(I am asking you this because I have set a different backup path in Ribbon WB.)

**Comment by APEbbers** [comment]
Hi,

copying the entire mod folder, saves you indeed from installing all wb's.
if the backup folder is located somewhere else, you select the same folder after install again.
You then can click on "restore" to select a backup file 
![image](https://github.com/user-attachments/assets/fd428bd7-6ccf-4a63-bd42-9d60263157ee)

**Comment by raindropsfromsky** [comment]
BTW my original feature request is to **autohide the ribbon**, just like how the side panels autohide at panel, and appear when we hover the mouse over that area.

In response, you have described multiple issues in FC which inhibit this feature.

```
In FC 1.0 you can toggle the overlay function by pressing F3. The ribbon
should dissapear to the topleft corner of the view window.
I have disabled the AutoShow function in FC1.0 because entering the Ribbon
with the mouse interferes with this overlay function an makes the ribbon
disappear.

You can click on the pinbutton to enable the AutoHide function. When you
click on a tab, the ribbon appears. When you leave the ribbon with the
mouse the ribbon fold again.

Personally, I dont want the ribbon to disapear completly like the panels
because it is a replacement for the toolbars and not an another panel. But
I've not figured out how the exclude the ribbon from the overlay function.

```
Are these issues filed in the FC GitHub? If not, this feature will remain illusive!

I checked the FC GitHub, and found [this](https://github.com/FreeCAD/FreeCAD/issues/17500) bug filed by you. I am not sure if that covers all the issues you have listed above.

Personally, I would like a plain _autohide_ feature based on the movement of the mouse, rather than hitting F3 or clicking on a [...] button. Is that possible?

BTW both methods you described do not work for me at all:
- The F3 button makes the whole screen blank
- The ribbon does not collapse at all, and I do not see a [...] button that restores it.

**Comment by APEbbers** [comment]
Hi, I've found a fix and a solution for the next release.

From version 1.4.12: When the Ribbon disappeared due to the overlay toggle, just restart FreeCAD and the ribbon will be restored.

In the next release i will implement a custom function to toggle overlay for all panel except the Ribbon.

**Comment by raindropsfromsky** [comment]
Did not work. 

I am using Ribbon 1.4.12 on FC dev 1.1.0.
I hovered the mouse on Ribbon and pressed F3. The ribbon did not disappear, but the windows bar disappeared. (It looked like "full screen" mode.)

I restarted FC. But now the entire screen is blank!
![image](https://github.com/user-attachments/assets/b1c66c69-318b-4d9e-931e-cae14023b05d)

I see only a couple of icons in the status bar.

I tried to press F3, F4 etc, but nothing works!

**Comment by APEbbers** [comment]
In the top left corner you should see:
![image](https://github.com/user-attachments/assets/8455443c-59b3-4701-ad98-c029bcc61134)
after you have pressed F3. 

For the tree:
![image](https://github.com/user-attachments/assets/fb4ccf5d-1ad1-47f2-a2f8-d7573693d948)
It is strange, I've tested the function on both Linux and Windows and it works normally.

**Comment by raindropsfromsky** [comment]
Nope!

It's a completely blank window as my screenshot shows.
Just to uynderstand this properly, where should my mouse be when I press F3 or F4?
I may have placed it in a wrong place.

Will I have to start all over again? (Except for saving the Mod folder...)

**Comment by APEbbers** [comment]
Try if it helps when you add an file named "ADDON_DISABLED" in the folder of the ribbon.
Just create a text file and rename it. (No extension!)

It will disable the add-on.
![image](https://github.com/user-attachments/assets/667f0217-d4cb-428f-b2af-d10e6526c5fe)

**Comment by raindropsfromsky** [comment]
I deleted all files in my FC installation folder, then downloaded the latest 7z file, and unzipped its contents into the installation folder.

Then I added a file without extension to this path:
![image](https://github.com/user-attachments/assets/1fefa46d-a1d0-4db0-8592-8c7e1471de84)
Then I started FC.

FC still has a blank UI.
![image](https://github.com/user-attachments/assets/67e5747b-2e41-43cb-9f6d-a9c0a4882456)

**Comment by APEbbers** [comment]
Then it isn't the ribbon. Can you try creating a shortcut with "--safe-mode" at the ned of target and see if FC will start?:
If it does, go to preferences and click reset. See if that helps

![image](https://github.com/user-attachments/assets/381170b4-859b-4b86-8d6d-4186b0891d98)

**Comment by raindropsfromsky** [comment]
I did the following:
1. Changed the properties of the FC icon in the taskbar to add "--safe-mode".
2. Ran FC in safe mode
3. Selected **Preferences**, and clicked on **Reset ALL** (not just the **general page** or **general group**).
4. Closed FC (in safe mode).
5. Again launched FC in safe mode, and everything looked fine (all the customization was lost, as expected).

6. Removed the safe mode from the properties of the taskbar icon
7. Launched FC in normal mode.

It reverts to the blank screen!
